src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -627,7 +628,10 @@ def process_info_file(cp_elem, info_file):
 
 class ZincCompile(BaseZincCompile):
   """Compile Scala and Java code to classfiles using Zinc."""
-
+  deprecated_module(
```

Comment:
I think that rather than deprecating the module (which will trigger for every import), you're going to want to deprecate the choice here maybe: https://github.com/pantsbuild/pants/blob/fd346c53ccbe22ec6076ef158383ca4bd8e24f66/src/python/pants/backend/jvm/subsystems/jvm_platform.py#L35-L48